sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ChangeFeedProcessorContext.java

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.util.Beta;
+
+import java.util.function.BiConsumer;
+
+/**
+ * Encapsulates properties which are mapped to a batch of change feed documents
+ * processed when {@link  ChangeFeedProcessorBuilder#handleAllVersionsAndDeletesChanges(BiConsumer)}
+ * lambda is invoked.
+ * */
+@Beta(value = Beta.SinceVersion.V4_50_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public interface ChangeFeedProcessorContext {
```

Comment:
@milismsft - I was looking at a possible version for a lease token / lease and it would map to a `PARTITION_KEY_BASED_LEASE` or an `EPK_RANGE_BASED_LEASE`. Curious to know what supportability benefits do we get by exposing it.

Reply:
Do we already have an internal enum for the versioning? 
Are we planning to expose this?

Reply:
Internally we have the `LeaseVersion`.

Reply:
Synced up offline with @FabianMeiswinkel - I'll be leaving out the version since it is part of the lease.